### DISCUSSION TITLE
Help me understand the reason behind the volume/backup size

### DISCUSSION THREAD

**Discussion by abrondijk** [longhorn/longhorn#3600] [Q&A]
## Environment
In case required, my setup info from the issue template:

 - Longhorn version: v1.2.3
 - Kubernetes version: v1.22.6
 - Node config
   - OS type and version: Mixture of Ubuntu server 21.10 and 20.04.3
   - CPU per node: Mixture of 4 and 8
   - Memory per node: Mixture, between 2 to 16GB
   - Disk type: Mixture of consumer sata SSDs and enterprise NVMe 
   - Network bandwidth and latency between the nodes: 1Gb/s per node over a 10Gb/s link
 - Underlying Infrastructure: Baremetal

## Question
Could someone explain the reason for the size of a volume's actual size and its backup size?
It's a volume with a size of 50GiB, an actual size of 19.2GiB, but if I exec into the pod its attached to:
```bash
/prometheus $ du -sh .
2.2G    .
```
Now from what I understand of the docs, the actual size includes snapshots, of which there are 2 (created by a backup job). This could explain some increase, but not an increase of ten times, especially as the pod in question (prometheus) only adds data and doesn't delete any.

I did find https://github.com/longhorn/longhorn/issues/951 and this https://github.com/longhorn/longhorn/issues/2510, could it be related to that? If it is, wouldn't the size of a backup be almost equal to the size that is shown in the pod?
This does not appear to be the case, as longhorn reports the 2 existing backups to be 6.9GiB and 9.8GiB respectively, while the MinIO instance its backing up to, reports a bucket usage of 5.4GiB.

Is there something wrong with my setup or am I fundamentally misunderstanding longhorn?

**Comment by derekbit** [comment]
> Could someone explain the reason for the size of a volume's actual size and its backup size?
It's a volume with a size of 50GiB, an actual size of 19.2GiB, but if I exec into the pod its attached to:

Creating files and deleting them will increase the actual replica size, and the `du -sh` in the pod can only calculate the total size of files in the filesystem. 

> This does not appear to be the case, as longhorn reports the 2 existing backups to be 6.9GiB and 9.8GiB respectively, while the MinIO instance its backing up to, reports a bucket usage of 5.4GiB.

Before running backup job, Longhorn takes a snapshot of the replica and backs it up to the remote repository.
Additionally, the backup job compresses and deduplicates the 2MiB blocks, so the size on backup ui is different from the size you see in the remote repository.

**Reply by abrondijk** [reply]
> Before running backup job, Longhorn takes a snapshot of the replica and backs it up to the remote repository. Additionally, the backup job compresses and deduplicates the 2MiB blocks, so the size on backup ui is different from the size you see in the remote repository.

Okay, so because longhorn backups blocks rather than files, the backup size can be larger than the filesize. That would certainly explain things. Thank you!
